Model/CustomItemExportSchedulerModel.php

```diff
@@ -0,0 +1,294 @@
+        $this->customItemRepository            = $customItemRepository;
+        $this->eventDispatcher                 = $eventDispatcher;
+    }
+
+    public function getRepository(): CustomItemExportSchedulerRepository
+    {
+        /** @var CustomItemExportSchedulerRepository $repo */
+        $repo = $this->em->getRepository(CustomItemExportScheduler::class);
+
+        return $repo;
+    }
+
+    /**
+     * @throws \Doctrine\ORM\ORMException
+     * @throws \Doctrine\ORM\OptimisticLockException
+     */
+    public function saveEntity(int $customObjectId): CustomItemExportScheduler
+    {
+        $customItemExportScheduler = new CustomItemExportScheduler();
+        $customItemExportScheduler
+            ->setUser($this->userHelper->getUser())
+            ->setScheduledDateTime(new DateTimeImmutable('now', new DateTimeZone('UTC')))
+            ->setCustomObjectId($customObjectId);
+
+        $this->em->persist($customItemExportScheduler);
+        $this->em->flush();
+
+        return $customItemExportScheduler;
+    }
+
+    public function processDataAndGetExportFilePath(CustomItemExportScheduler $customItemExportScheduler): string
+    {
+        $scheduledDateTime = $customItemExportScheduler->getScheduledDateTime();
+        $fileName          = 'custom_items_export_'.$scheduledDateTime->format(self::EXPORT_FILE_NAME_DATE_FORMAT).'.csv';
+
+        $filePath = $this->exportHelper->getValidExportFileName($fileName, 'custom_item_export_dir');
+
+        $this->filePath = $filePath;
+
+        $this->processCustomItemData($customItemExportScheduler);
+
+        return $this->exportHelper->zipFile($this->filePath, 'custom_item_exports.csv');
+    }
+
+    private function processCustomItemData(CustomItemExportScheduler $customItemExportScheduler): void
+    {
+        $customObjectRepo = $this->em->getRepository(CustomObject::class);
+
+        $customObject = $customObjectRepo->find($customItemExportScheduler->getCustomObject());
+
+        $customFields = $customObject->getCustomFields()->toArray();
+
+        $this->addExportFileHeaderToCsvFile($customFields);
+
+        $this->addCustomItemsToCsvFile($customFields, $customObject);
+    }
+
+    /**
+     * @param array<CustomField> $customFields
+     *
+     * @return array<string>
+     */
+    private function getCSVHeader(array $customFields): array
+    {
+        $header = ['customItemId', 'customItemName'];
+
+        foreach ($customFields as $customField) {
+            $header[] = $customField->getId();
+        }
+
+        $header[] = 'linkedContactIds';
+
+        return $header;
+    }
+
+    /**
+     * @param array<CustomField> $customFields
+     */
+    private function addCustomItemsToCsvFile(array $customFields, CustomObject $customObject): void
+    {
+        $offset = 0;
+        $result = true;
+
+        $handler = @fopen($this->filePath, 'ab+');
+
+        while ($result) {
```

Comment:
The cyclomatic complexity is too high here. We should extract some of this into private methods.